### PR TITLE
Fix/validation err msg

### PR DESCRIPTION
**Pull request by yuki-o1o5**
## Main Proposed Changes
Add Validation, error message and show the sign if The transaction is paid or not
#### Code changes
- Add Validation
- Add error message
- Add the sign to transaction card
- Adjust Responsive
- Add Goback button

#### Related Issues
#52

### CODE REVIEW

File: frontend/src/views/pages/PaymentPage.tsx [hunk 1]
```diff
@@ -65,27 +85,29 @@ export const PaymentPage = () => {
...
-  };
+  });
 
   console.log(error);
 
   const insertExpense = async () => {
+    if (description === "") {
+      setError("Description cannot be empty");
+      return false;
+    }
```
Comment (Ak1t0-Dev) [inline frontend/src/views/pages/PaymentPage.tsx:103]: please also check if the description is not a blank like " " and "mulitple spaces".

File: frontend/src/views/pages/PaymentPage.tsx [hunk 2]
```diff
@@ -418,17 +471,10 @@ const StyledOutlinedInput = styled(OutlinedInput)`
 `;
 
 const StyledOutlinedNumberInput = styled(StyledOutlinedInput)`
-  && input::-webkit-outer-spin-button,
-  && input::-webkit-inner-spin-button {
```
Comment (Ak1t0-Dev) [inline frontend/src/views/pages/PaymentPage.tsx:422]: You thought it is better to display arrows in an amount input? I just want to make sure and It's up to you.

// before
Screenshot 2023-06-24 at 8 40 39 PM

// after
Screenshot 2023-06-24 at 8 32 25 PM

**Review by Ak1t0-Dev — COMMENTED** [review]
Please read all the comments first and revise the pointed parts, as some of them are scattered across different sections.

File: frontend/src/views/components/AccountInput.tsx [hunk 3]
```diff
@@ -63,7 +76,9 @@ export const AccountInput = ({
       data: updatedMetaData,
     });
     if (error) {
-      console.error("Error updating user email and metadata:", error);
+      setErrorUpdateUserInfo(ERROR_CHANGE_ACCOUNT_INFO);
+      console.error(ERROR_CHANGE_ACCOUNT_INFO);
```
[the remark this answers is not in the document]
Comment (yuki-o1o5) [reply frontend/src/views/components/AccountInput.tsx:80]: I already removed it before commit.

File: frontend/src/App.css
[the remark this answers is not in the document]
Comment (yuki-o1o5) [reply frontend/src/App.css:1]: Okay

File: frontend/src/constants/messages.ts [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+export const ERROR_EMAIL = "Please enter valid email format";
+
+export const ERROR_PASSWORD =
+  "6+ characters, with at least one upper and a digit";
+
+export const ERROR_PASSWORDCONF = "Please confirm your password";
+
+export const ERROR_USER_NOTFOUND = "User not found";
+
+export const ERROR_USER_EXIST = "User already exists";
```
[the remark this answers is not in the document]
Comment (yuki-o1o5) [reply frontend/src/constants/messages.ts:28]: Okay! Thanks

File: frontend/src/views/components/AccountInput.tsx [hunk 5]
```diff
@@ -91,19 +108,22 @@ export const AccountInput = ({
           redirectTo: "http://localhost:3000/passwordReset",
```
[the remark this answers is not in the document]
Comment (yuki-o1o5) [reply frontend/src/views/components/AccountInput.tsx:108]: Okay!!

File: frontend/src/views/pages/PaymentPage.tsx (hunk 2 above)
Comment (yuki-o1o5) [reply frontend/src/views/pages/PaymentPage.tsx]: Sorry. Removed arrows.

File: frontend/src/views/pages/SignUpPage.tsx [hunk 6]
```diff
@@ -178,13 +179,19 @@ export const SignUpPage = () => {
...
           <ButtonWrapper>
             <FormButton title="register" />
           </ButtonWrapper>
-          {authError && <ErrorText>{authError}</ErrorText>}
+          {authError && <ErrorTextSignup>{authError}</ErrorTextSignup>}
+          {authSuccess && (
+            <SuccessTextSignup
+              dangerouslySetInnerHTML={{ __html: authSuccess }}
+            />
+          )}
```
[the remark this answers is not in the document]
Comment (yuki-o1o5) [reply frontend/src/views/pages/SignUpPage.tsx:194]: Thanks, Fixed it.

File: frontend/src/views/pages/PaymentPage.tsx (hunk 1 above)
Comment (yuki-o1o5) [reply frontend/src/views/pages/PaymentPage.tsx:103]: Okay. I add .trim() method.

File: frontend/src/constants/messages.ts [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+//Account Page
+export const deleteMsg = (
+  callback: (string: string) => void,
+  string: string
+): void => {
+  setTimeout(() => {
+    callback(string);
+  }, 10000);
+};
```
[the remark this answers is not in the document]
Comment (yuki-o1o5) [reply frontend/src/constants/messages.ts:38]: Changed file name to textUtils and moved inside of utils directory.
Removed deleteMsg.

File: frontend/src/views/components/AccountInput.tsx [hunk 8]
```diff
@@ -161,10 +185,19 @@ export const AccountInput = ({
               <SubButton title={"edit"} onClick={handleEdit} />
             )}
           </ButtonContainer>
+          {errorUpdateUserInfo && (
+            <ErrorText
+              dangerouslySetInnerHTML={{ __html: errorUpdateUserInfo }}
```
[the remark this answers is not in the document]
Comment (yuki-o1o5) [reply frontend/src/views/components/AccountInput.tsx:190]: Add the followings. 

export const addNewLinesAfterPunctuation = (message: string) => {
  return message.replace(/([.,])/g, "$1\n");
};

div {
  white-space: pre-wrap;
}

File: frontend/src/constants/messages.ts [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+    /([.,])/g,
+    "$1<br />"
+  );
+
+//Signup Page
+export const SUCCESS_SIGNUP =
+  "Email Sent! Please check your inbox for further instructions on starting this app.".replace(
+    /([.,])/g,
+    "$1<br />"
+  );
```
[the remark this answers is not in the document]
Comment (yuki-o1o5) [reply frontend/src/constants/messages.ts:13]: Thanks. Fixed it

File: frontend/src/constants/messages.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (Ak1t0-Dev) [reply frontend/src/constants/messages.ts]: I mean
for constants such as messages → constants directory
for common functions → utils directory (data formating functions or deleteMsg above)
so you don't have to change the file name and move it if it is only messages are written.
// constants > mesages.ts
export const ERROR_USER_NOTFOUND = "User not found";
// utils > textUtils.ts
export const addNewLinesAfterPunctuation = (message: string) => {
return message.replace(/([.,])/g, "$1\n");
};

File: frontend/src/constants/regexPattern.ts [hunk 10]
```diff
@@ -2,15 +2,3 @@ export const passwordRegex =
   /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[a-zA-Z]).{6,}$/;
 
 export const emailRegex = /\S+@\S+\.\S+/;
```
Comment (Ak1t0-Dev) [inline frontend/src/constants/regexPattern.ts:4]: Could you move this file to the utils directory?

**Review by Ak1t0-Dev — APPROVED** [review]

**Pull request merged**